aws_xray_sdk/core/models/entity.py

```diff
@@ -256,36 +257,42 @@ def get_origin_trace_header(self):
     def serialize(self):
         """
         Serialize to JSON document that can be accepted by the
-        X-Ray backend service. It uses jsonpickle to perform
-        serialization.
+        X-Ray backend service. It uses json to perform serialization.
         """
         try:
-            return jsonpickle.encode(self, unpicklable=False)
+            return json.dumps(self.to_dict(), default=str)
         except Exception:
             log.exception("got an exception during serialization")
```

Comment:
We should log the exception as well to understand what went wrong. Can be at debug level.

Reply:
Updated

Reply:
I meant log the actual exception message and stack trace at debug level.